File: 0202-happy-number/0202-happy-number.cpp
class Solution {
public:
    bool isHappy(int n) {
        if(n==1) return true;
        // if(n<10) return false;
        set<int> x;
        while(n!=1){
            x.insert(n);
            int z=n;
            int c=0;
            while(z!=0){
                c=c+pow(z%10,2);
                z=z/10;
            }
            if(x.find(c)!=x.end()) return false;
            if(c==1) return true;
            n=c;
        }
        return true;
    }
};

File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        int f = n,s = n;
        do{
            f = square(square(f));
            s = square(s);
            if(f == 1 || s == 1) return true;
        }while(s!=f);
        
        // if(s == 1 || f==1) return true;
        return false;
    }
    public int square(int n){
        int ans = 0;
        while(n>0){
            int rem = n%10;
            ans = ans + (rem*rem);
            n = n/10;
        }
        return ans;
    }
}